### ISSUE TITLE
unit tests failing due to xray not available locally

### ISSUE THREAD

**Issue by dspaxton** [aws-samples/cookiecutter-aws-sam-python#4]
=========================================================== test session starts ===========================================================
platform darwin -- Python 3.6.5, pytest-3.5.1, py-1.5.3, pluggy-0.6.0 -- /Users/paxtod/.local/share/virtualenvs/sample-project--BvM5prf/bin/python
cachedir: .pytest_cache
rootdir: /Users/paxtod/WorkDocs/code/sample-project, inifile:
plugins: cov-2.5.1
collected 1 item

tests/unit/test_handler.py::test_lambda_handler FAILED                                                                              [100%]

================================================================ FAILURES =================================================================
___________________________________________________________ test_lambda_handler ___________________________________________________________

apigw_event = {'body': '{ "test": "body"}', 'headers': {'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,...ept-Language': 'en-US,en;q=0.8', 'Cache-Control': 'max-age=0', ...}, 'httpMethod': 'POST', 'path': '/examplepath', ...}

    def test_lambda_handler(apigw_event):

>       ret = app.lambda_handler(apigw_event, "")

tests/unit/test_handler.py:89:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
first_function/app.py:27: in lambda_handler
    message = get_message()
../../../.local/share/virtualenvs/sample-project--BvM5prf/lib/python3.6/site-packages/aws_xray_sdk/core/recorder.py:359: in wrapper
    meta_processor=None,
../../../.local/share/virtualenvs/sample-project--BvM5prf/lib/python3.6/site-packages/aws_xray_sdk/core/recorder.py:367: in record_subsegment
    subsegment = self.begin_subsegment(name, namespace)
../../../.local/share/virtualenvs/sample-project--BvM5prf/lib/python3.6/site-packages/aws_xray_sdk/core/recorder.py:231: in begin_subsegment
    segment = self.current_segment()
../../../.local/share/virtualenvs/sample-project--BvM5prf/lib/python3.6/site-packages/aws_xray_sdk/core/recorder.py:215: in current_segment
    entity = self.get_trace_entity()
../../../.local/share/virtualenvs/sample-project--BvM5prf/lib/python3.6/site-packages/aws_xray_sdk/core/recorder.py:316: in get_trace_entity
    return self.context.get_trace_entity()
../../../.local/share/virtualenvs/sample-project--BvM5prf/lib/python3.6/site-packages/aws_xray_sdk/core/context.py:93: in get_trace_entity
    return self.handle_context_missing()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <aws_xray_sdk.core.context.Context object at 0x103d352b0>

    def handle_context_missing(self):
        """
            Called whenever there is no trace entity to access or mutate.
            """
        if self.context_missing == 'RUNTIME_ERROR':
            log.error(MISSING_SEGMENT_MSG)
>           raise SegmentNotFoundException(MISSING_SEGMENT_MSG)
E           aws_xray_sdk.core.exceptions.exceptions.SegmentNotFoundException: cannot find the current segment/subsegment, please make sure you have a segment open

../../../.local/share/virtualenvs/sample-project--BvM5prf/lib/python3.6/site-packages/aws_xray_sdk/core/context.py:118: SegmentNotFoundException
------------------------------------------------------------ Captured log call ------------------------------------------------------------
context.py                 117 ERROR    cannot find the current segment/subsegment, please make sure you have a segment open
======================================================== 1 failed in 1.07 seconds =========================================================

**Comment by heitorlessa** [comment]
Hey - Thanks for that @dspaxton, can you try running `make test` to see if it works?

If it does we need to update the README file to use the following:

```
AWS_XRAY_CONTEXT_MISSING=LOG_ERROR pipenv run python -m pytest --cov . --cov-report term-missing --cov-fail-under $(CODE_COVERAGE) tests/ -v
```

**Comment by dspaxton** [comment]
@heitorlessa make test appears to work fine. 

Executing the proposed line for README results in this:

`AWS_XRAY_CONTEXT_MISSING=LOG_ERROR pipenv run python -m pytest --cov . --cov-report term-missing --cov-fail-under $(CODE_COVERAGE) tests/ -v`
`zsh: command not found: CODE_COVERAGE`
`usage: pytest.py [options] [file_or_dir] [file_or_dir] [...]`
`pytest.py: error: argument --cov-fail-under: invalid int value: 'tests/'`

**Comment by heitorlessa** [comment]
Great! We have two options for now and if you want to make a PR to help
bonus points for you hehe:

- We can update the README with that line as Windows users may not have
Make installed
- Alternatively we can update the unit test to cover this this too

**Comment by h1f1x** [comment]
@heitorlessa as I remember we "fixed" this with the yesterdays PR #5 ! So your could close this issue!

**Comment by heitorlessa** [comment]
That's right! Resolving this now and thanks @h1f1x for the testing too!
